### ISSUE TITLE
Inconsistent window state

### ISSUE THREAD

**Issue by StyxOfDynamite** [MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-#2868]
**OS:**
>macOS 10.13.1 (17B48)

**Issue Descriptions:**
Full player UI is unusable
- No scroll functionality
- Clicking UI elements does not fire any human noticeable interaction.
- Returning to 'Mini-player' from the `Window` menu is not possible.

**Steps to Reproduce:**
When running the player in 'Mini-player' if I want to change album, station, playlist
I click the icon to launch the main app.

**screenshots.**
![begin-state](https://user-images.githubusercontent.com/946550/32832583-e6444200-c9f3-11e7-97b1-7cb5506373ac.png)
![end-state](https://user-images.githubusercontent.com/946550/32832584-e65efc26-c9f3-11e7-8584-3997536ddd69.png)
<img width="233" alt="end-state-window-menu" src="https://user-images.githubusercontent.com/946550/32832585-e6788240-c9f3-11e7-9496-8be64d594fdc.png">

**Comment by MarshallOfSound** [comment]
> Clicking UI elements does not fire any human noticeable interaction.

Am I missing something here? Music is clearly playing in that screenshot so you must have clicked something successfully 🤔

**Comment by StyxOfDynamite** [comment]
>Am I missing something here? Music is clearly playing in that screenshot so you must have clicked something successfully 🤔

Steps to Reproduce:
When running the player in 'Mini-player' if I want to change album, station, playlist
I click the icon to launch the main app.

**Comment by BrianAllred** [comment]
There's an icon in the bottom right of the mini-player that brings up the full player... 

![mini](https://user-images.githubusercontent.com/5847511/32848358-ddcac8b2-c9f1-11e7-9ebc-3b812fda40d9.png)

**Comment by StyxOfDynamite** [comment]
>Steps to Reproduce:
When running the player in 'Mini-player' if I want to change album, station, playlist
I click the icon to launch the main app.
### I click the icon to launch the main app.

I understand you get a lot of issues, and possibly english isn't your primary language. But please respond with something useful.

Here's is how I would suggest you approach this ticket.

Read the issue.
Follow the steps
Respond.
I'd rather see

>Followed replication steps, unable to replicate on latest build, close issue

or

> Followed steps, bug noted, bug tagged, thanks.

**Comment by jostrander** [comment]
@BrianAllred pointed out exactly what you are looking for. The mini player only allows a subset of actions. You need to click the button he pointed out in order to change playlists, pick a new album, etc. Quite frankly your tone is unacceptable and won't be tolerated.

**Comment by jostrander** [comment]
Also: Follow replication steps, unable to reproduce on latest release.

**Comment by StyxOfDynamite** [comment]
@jostrander - please note this is the first time, I've indicated that any response to this thread referred explicitly to somebody.

>A good bug report shouldn't leave others needing to chase you up for more information.

I can't help if others feel the need to confirm what hasn't been reported.

>Music is clearly playing in that screenshot so you must have clicked something successfully

In my replication steps, I described what state the player was in, and what button I clicked to cause the bug.

Now let's discuss 'tone'. tone, incase anyone is unsure is how **you** hear what __I__ wrote.

I didn't claim that the music wasn't playing, i said after I clicked a button I couldn't trigger any noticeable interaction.

I could tell that music was playing, I was listening to it, and it didn't seem relveant

I noted that the unrelated assumption made was correct, and explained that it was only because I wanted to do something that wasn't possible in the current UI that I tried to do it in the main UI.

>but if i wanted the same song to continue playing, I wouldn't be trying to do any of the actions I listed in the steps.

This gives quite clear context as to why i was trying to launch the full player app,

If I had expressed any sort of tone, it would have been frustration directed at myself for not being understood, but that happens a lot to me for some reason.

>I want to change album, station, playlist not keep listening and look at the Full player

Had you seen, what I had seen, when running the two click replication steps you'd understand why I didn't feel like looking at the now unusable place and listening to the same music.

When 5 hours later, bearing in mind by this point I could still replicate the issue on latest and I got a notification I was admittedly surprised to see it telling me which button I needed to click on to cause the error I had reported.

I repeated the steps needed to trigger the issue and asked politely that someone, respond with something useful. I suggested how I would handle the issue, somewhat unrelated again but I fix and resolve bugs for a living and would have been over the moon to get a two click replication bug.

You @jostrander then chose to respond to me.

At this point, I've just been told where to find the button I mentioned as the last part of the replication steps.

>xyz...pointed out exactly what you are looking for. The mini player only allows a subset of actions. You need to click the button he pointed out in order to change playlists, pick a new album, etc. 

He didn't he pointed out the button required to get into a strange situation, that I was trying to help sort out,

You then described the very reasons I wanted to click the button to begin with, finally I thought were were going to start looking at a PR.

Thank you for doing so :)

Full disclosure now.

When I'm not on the receiving end of being told that

>Frankly your tone is unacceptable and won't be tolerated.

When I'm not fixing technical bugs that require significantly more than two mouse clicks in order to replicate them.

And when I'm not struggling with my inability to comprehend or discern 'tone' from people talking to me face to face .

'm working with a few people and some local charities to make Open Source communities more accessible to others who suffer with [schizoaffective disorder](https://www.mayoclinic.org/diseases-conditions/schizoaffective-disorder/symptoms-causes/)

I'm actually pitching a talk at a UK PHP conference on pretty much exactly why it is very difficult to work in an industry when you experience

- Delusions — having false, fixed beliefs, despite evidence to the contrary
- Hallucinations, such as hearing voices or seeing things that aren't there
- Symptoms of depression, such as feeling empty, sad or worthless
- Periods of manic mood or a sudden increase in energy with behavior that's out of character
- Impaired communication, such as only partially answering questions or giving answers that are completely unrelated
- Impaired occupational, academic and social functioning
- Problems with managing personal care, including cleanliness and physical appearance

Often leading to common complications such as

- Suicide, suicide attempts or suicidal thoughts
- Social isolation
- Family and interpersonal conflicts
- Unemployment
- Anxiety disorders
- Developing alcohol or other substance abuse problems
- Significant health problems
- Poverty and homelessness

When you get stressed and how I've for the most part avoided the most nuclear of options on the above list.

**Comment by StyxOfDynamite** [comment]
@MarshallOfSound 

Made a video, still happens for me, sorry steps weren't clear, https://youtu.be/fHyEgA_p2X8

**Comment by BrianAllred** [comment]
Apologies for misunderstanding you; while English is my first (and admittedly, only) language, I have a difficult time understanding the way you write.

While you may not mean it this way, the manner in which you write comes across as condescending and patronizing, which makes people defensive and uncomfortable. Suggesting how developers approach tickets can definitely come across the wrong way. This, combined with the unclear reproduction steps, is probably where a lot of our dismissiveness is coming from.

If I may make a few suggestions for future issues:

* **Include verifiable, repeatable steps to reproduce the problem**, preferably numbered and definitively labeled. For example:
1. Launch app.
2. Play song.
3. Click icon to switch to mini player. (Possibly include screenshot showing the icon)
4. Click icon to switch to full player.
5. Click artist picture.

* **Include a screenshot of the problem.** The screenshot you initially provided didn't look out of the ordinary, which is one reason I was confused.

* **Always assume the developer wants to help.** As a professional and hobby developer myself, I want to fix all issues with my software, but I also don't want people to invent problems that aren't there. So if a developer sounds confused about an issue you have, ask yourself what you can do to make the problem more clear. Remember, the more concise and simple the explanation, the more clarity and focus you'll bring the the issue.

So now, after reviewing your video, I'd agree that you have discovered a bug. @jostrander and @MarshallOfSound are the bosses around here, so one of them will need to review it.

**Comment by StyxOfDynamite** [comment]
@BrianAllred - I'm sorry if thats the way I came across. I'm happy to open an issue suggesting improvements to the contributing.md file that would mitigate the way I speak being an issue.

Also it, as per the title, it is inconsistent, other variations I noticed while trying to get a reliable set of instruction are exactly as I describe in the original issue where opening the full player from the mini player UI result in the UI being non-clickable, the side menu doesn't ever dismiss, meaning clicking on artist icons, or menu items like 'Radio', 'Playlists' doesn't do anything.

From the same perspective, i could provide an analysis as to why an emojii implying significant thought is required to understand me is equally if not more patronising, condescending and not helpful. It's immediately put me on the back foot, and any frustration I may have expressed, has been wrongly identified.  I was not asked to describe the issue in any more detail, so it wasn't clear the replication steps were not enough. I was told that it wasn't a problem and that the button to trigger the issue was what i needed to be clicking.

Perhaps a standard for language to describe the issue would be something worthwhile, I'd even submit an example PR are you familiar with gherkin syntax?

```
Scenario:  Listening to music in the mini-player
  Given I want to listen to a song not in the current playlist
  When I click the launch full app button
  Then the full-screen UI window state is incoisisntent
```

**Comment by jostrander** [comment]
This has blown way out of proportion. Clearly there is an issue at hand here, and I believe it's actually been reported before with a different set of criteria, maybe #2269. I followed your simplistic replication instructions, which was basically:

> 1. Use mini player
> 2. Switch to normal player, notice it's not responding to user input.

Unfortunately I wasn't able to reproduce the issue you have, which isn't an uncommon occurrence. There are many environmental variables at play as to why that could be. Where we got off on the wrong foot was when you stated:

> Here's is how I would suggest you approach this ticket.

Which seemed rather condescending considering this is free software and no one has to make time to help you out with it, bugs or not. I'd consider adding your feedback on #2269 and we'll update anything we have related to that there.

**Comment by StyxOfDynamite** [comment]
@jostrander The only out of hand bit was your response. this is free software and no one has to make time to help you out with it.

You are right :)

>It is free software and I don't have time to help you out with it, and what little time I did give up was largely wasted

**Comment by StyxOfDynamite** [comment]
Removed stuff not relevant to helping people fix their code.

**Comment by MarshallOfSound** [comment]
Alright folks, from what I can see here things got quite heated for whatever reason so this is the final word.

1. @StyxOfDynamite Although potentially not your intention, your tone definitely came across as condescending and actually when I first saw that comment I just tossed my phone away because I didn't want to respond to it.  People put a lot of time into this project and it's always good to err on the side of appreciating that instead of telling us how to do our jobs.
2. Initially I believe all of us missed the actual issue being described here, this IMO is due to the _improvable_ way the original issue was described.  "inconsistent" does not describe the actual issue here, a better descriptor would have been "freezes", "locks up" or "becomes unresponsive to user input".  Our issue template asks for steps to reproduce but maybe we should also ask for a _detailed_ description of what is wrong including symptoms.
3. @jostrander is correct in their decision that this issue should be merged into #2269 😄 

And now have a bunny --> 🐰
